### PR TITLE
doc: add a page on profiling

### LINKED ISSUE #11389: Document memory profiling for neard
Turns out that getting neard run with a memory profiler is a not very easy! I had some success with bytehound. We should document the process and choices so that other people need not to rediscover this all themselves next time memory profiling comes up.

### PR DESCRIPTION
**Pull request by nagisa**
This includes both the somewhat well known information on how to use perf as well as text on bytehound, which is the only tool that seems to work for memory profiling neard.

Fixes #11389

### CODE REVIEW

File: docs/practices/workflows/profiling.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+<blockquote style="background: rgba(255, 200, 0, 0.1); border: 5px solid rgba(255, 200, 0, 0.4);">
+
+Do note that you will need about twice the amount of RAM as the size of the input file in order to
+load it successfully.
+
+</blockquote>
+
+Once enough profiling data has been gathered, terminate the program. Use the `bytehound` CLI tool
+to operate on the profile. I recommend `bytehound serve` over directly converting to e.g. heaptrack
```
Comment (saketh-are) [inline docs/practices/workflows/profiling.md:93]: I think this is a typo; the command is `bytehound server`

File: docs/practices/workflows/profiling.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+### Viewing the profile
+
+<blockquote style="background: rgba(255, 200, 0, 0.1); border: 5px solid rgba(255, 200, 0, 0.4);">
+
+Do note that you will need about twice the amount of RAM as the size of the input file in order to
+load it successfully.
+
+</blockquote>
+
+Once enough profiling data has been gathered, terminate the program. Use the `bytehound` CLI tool
```
Comment (saketh-are) [inline docs/practices/workflows/profiling.md:92]: How are you terminating neard? If I use SIGTERM or SIGINT I get no output file from the profiler. I do see an output `.dat` if I run a command which terminates on its own (e.g. `env LD_PRELOAD=/home/ubuntu/libbytehound.so ls`).

Comment (nagisa) [reply docs/practices/workflows/profiling.md:92]: Just ^C. For me the profile file gets written continuously throughout the execution…

Comment (nagisa) [reply docs/practices/workflows/profiling.md:92]: I think the reason why it didn't work for you were incomplete instructions and the fact that the libbytehound was not getting attached... let me update them instructions.

**Review by saketh-are — APPROVED** [review]
LGTM, thanks!

May be worth mentioning to use a release build of neard. I experienced consistent segfaults otherwise.

**Pull request merged**